src/test/java/com/direwolf20/buildinggadgets/common/building/region/RegionSpliteratorTest.java

```diff
@@ -0,0 +1,10 @@
+package com.direwolf20.buildinggadgets.common.building.region;
+
+/**
+ * Both spliterator and streams
+ */
+public class RegionSpliteratorTest {
+
+    //TODO spliterator tests
```

Comment:
This should contain:

- Tests that the sorting is correct
- Test that splitting does not produce positions twice
- Tests that the returned sizes are correct, even across splitting